src/driver/DriverFactory.ts

```diff
@@ -27,48 +8,59 @@ export class DriverFactory {
     /**
      * Creates a new driver depend on a given connection's driver type.
      */
-    create(connection: DataSource): Driver {
-        const { type } = connection.options
+    create(dataSource: DataSource): Driver {
+        const { type } = dataSource.options
+        return new (this.getDriver(type))(dataSource)
+    }
+
+    private getDriver(type: DataSource["options"]["type"]): DriverConstructor {
         switch (type) {
             case "mysql":
-                return new MysqlDriver(connection)
+            case "mariadb":
+                return require("./mysql/MysqlDriver").MysqlDriver
```

Comment:
```suggestion
                return await import("./mysql/MysqlDriver").default
```

Reply:
Mixing `import` and `require` is not a good choice in my eyes.

Reply:
can't use import without breaking the public api.

Reply:
> can't use import without breaking the public api.

What is the problem with dynamic imports?